spec-tests/src/test/java/org/candlepin/spec/entitlements/AutobindSpecTest.java

```diff
@@ -0,0 +1,1120 @@
+            "system.certificate_version", "3.2"))
+            .installedProducts(Set.of(new ConsumerInstalledProductDTO().productId(product.getId())
+            .productName(product.getName()))));
+        ApiClient guestClient = ApiClients.ssl(guest);
+
+        // create a hypervisor that needs 40 sockets and report the guest with it
+        String hypervisorUuid = StringUtil.random("uuid");
+        ConsumerDTO host = userClient.consumers().createConsumer(Consumers.random(owner)
+            .hypervisorId(new HypervisorIdDTO().hypervisorId(hypervisorUuid))
+            .facts(Map.of("virt.is_guest", "false", "cpu.cpu(s)", "4",
+            "cpu.cpu_socket(s)", "40")));
+        ApiClient hostClient = ApiClients.ssl(host);
+        hostClient.consumers().updateConsumer(host.getUuid(),
+            new ConsumerDTO().guestIds(List.of(new GuestIdDTO().guestId(guestUuid))));
+
+        assertThat(ownerApi.listOwnerPools(owner.getKey())).hasSize(4);
+        guestClient.consumers().autoBind(guest.getUuid());
+        assertThat(ownerApi.listOwnerPools(owner.getKey())).hasSize(5);
+
+        // heal should succeed, and hypervisor should consume 2 pools of 30 sockets each
+        assertThat(hostClient.consumers().listEntitlements(host.getUuid())).hasSize(2);
+        assertThat(guestClient.consumers().listEntitlements(guest.getUuid())).hasSize(1);
+
+        hostClient.consumers().unbindAll(host.getUuid());
+        guestClient.consumers().unbindAll(guest.getUuid());
+
+        // change the hypervisor to 70 sockets
+        hostClient.consumers().updateConsumer(host.getUuid(), new ConsumerDTO()
+            .facts(Map.of("virt.is_guest", "false", "cpu.cpu(s)", "4",
+            "cpu.cpu_socket(s)", "70")));
+
+        // heal should succeed, and hypervisor should consume 3 pools of 30 sockets each
+        guestClient.consumers().autoBind(guest.getUuid());
+        assertThat(hostClient.consumers().listEntitlements(host.getUuid())).hasSize(3);
+        assertThat(guestClient.consumers().listEntitlements(guest.getUuid())).hasSize(1);
+        assertThat(ownerApi.listOwnerPools(owner.getKey())).hasSize(5);
+    }
+
+    @Test
+    public void shouldAttachToAddonPoolWhenProductIsNotInstalled() {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        ApiClient userClient = ApiClients.basic(UserUtil.createUser(client, owner));
+        ProductDTO product1 = ownerProductApi.createProductByOwner(owner.getKey(),
+            Products.randomEng()
+            .attributes(List.of(
+            ProductAttributes.Addons.withValue("addon1"))));
+        PoolDTO pool1 = ownerApi.createPool(owner.getKey(), Pools.random(product1));
+        ProductDTO product2 = ownerProductApi.createProductByOwner(owner.getKey(), Products.randomEng());
+        PoolDTO pool2 = ownerApi.createPool(owner.getKey(), Pools.random(product2));
+
+        ConsumerDTO consumer = userClient.consumers().createConsumer(
```

Comment:
This block, too. There are three operations going on here, but the indentation implies only one.